spec/integration/puppet_x/puppetlabs/powershell_manager_spec.rb

```diff
@@ -58,6 +58,13 @@ def create_manager
         expect(first_pid).to eq(second_pid)
       end
 
+      it "should fail if the manger is created with a short timeout" do
+
+        manager.exit
```

Comment:
You should be able to do this a bit more directly like:

```ruby
expect {
  manager = PowerShellManager.new(manager_args, false, 0.01)
}.to raise_error(RuntimeError, /Failure waiting for PowerShell process \d+ to start pipe server/)
```

I was going to suggest you verify that there's no PID running as well, but that will be hard given how the code is currently written.

Reply:
Good point. We did verify that there's no process running around behind our backs, but we did that manually outside of ruby.

Reply:
It turned out to be difficult to externally ask the manager you get from PowerShellManager.new(), what it's PID is.

Reply:
This ended up working:

```ruby
it "should fail if the manger is created with a short timeout" do
  expect {
    manager = PuppetX::PowerShell::PowerShellManager.new(manager_args, false, 0.01)
  }.to raise_error do |e|
    expect(e).to be_a(RuntimeError)
    expected_error = /Failure waiting for PowerShell process (\d+) to start pipe server/
    expect(e.message).to match expected_error
    pid = expected_error.match(e.message)[1].to_i
    expect{Process.kill(0, pid)}.to raise_error(Errno::ESRCH)
  end
end
```